### PR TITLE
#200: Fixed reuse database setup

### LINKED ISSUE #200: Exception Schema already exists in PopulateTestEngine when reusing DB
```
- PopulateTestEngine_bd2f349c75:
    Traceback (most recent call last):
      File "/home/vagrant/.cache/pypoetry/virtualenvs/exasol-script-languages-container-tool-fEttNriu-py3.8/lib/python3.8/site-packages/exasol_integration_test_docker_environment/lib/base/stoppable_base_task.py", line 27, in run
        yield from task_generator
      File "/home/vagrant/.cache/pypoetry/virtualenvs/exasol-script-languages-container-tool-fEttNriu-py3.8/lib/python3.8/site-packages/exasol_integration_test_docker_environment/lib/base/timeable_base_task.py", line 120, in run
        yield from task_generator
      File "/home/vagrant/.cache/pypoetry/virtualenvs/exasol-script-languages-container-tool-fEttNriu-py3.8/lib/python3.8/site-packages/exasol_integration_test_docker_environment/lib/base/base_task.py", line 248, in run
        raise e
      File "/home/vagrant/.cache/pypoetry/virtualenvs/exasol-script-languages-container-tool-fEttNriu-py3.8/lib/python3.8/site-packages/exasol_integration_test_docker_environment/lib/base/base_task.py", line 239, in run
        task_generator = self.run_task()
      File "/home/vagrant/.cache/pypoetry/virtualenvs/exasol-script-languages-container-tool-fEttNriu-py3.8/lib/python3.8/site-packages/exasol_integration_test_docker_environment/lib/test_environment/database_setup/populate_data.py", line 42, in run_task
        raise Exception("Failed to populate the database with data.\nLog: %s" % cmd + "\n" + output.decode("utf-8"))
    Exception: Failed to populate the database with data.
    Log: cd /tests_data/enginedb_small; $EXAPLUS -c '172.18.0.2:8888' -x -u 'sys' -p 'exasol' -f import.sql -jdbcparam 'validateservercertificate=0'
    EXAplus 7.0.11 (c) EXASOL AG
    
    Tuesday, June 6, 2023 at 8:19:40 AM Coordinated Universal Time
    Connected to database DB1 as user sys.
    EXASolution 7.1.17 (c) EXASOL AG
    
    EXA: CREATE SCHEMA TEST;
    Error: [42500] schema TEST already exists (Session: 1767940639488868352)
    
    Disconnected.

```

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #200

### CODE REVIEW

File: exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py [hunk 1]
```diff
@@ -22,18 +22,14 @@
...
-    #       - only possible if the hash of exaslc also goes into the image hashes
-
     reuse_uploaded_container = luigi.BoolParameter(False, significant=False)
     release_goal = luigi.Parameter()
 
     def __init__(self, *args, **kwargs):
         self.test_environment_info = None
         super().__init__(*args, **kwargs)
+        print("TestRunnerDBTestTask_args",args)
+        print("TestRunnerDBTestTask_kwargs", kwargs)
```
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py:32]:
```suggestion
```

File: exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py [hunk 2]
```diff
@@ -70,7 +66,10 @@ def run_task(self):
         yield from self.upload_container(database_credentials,
                                          export_info,
                                          reuse_release_container)
-        yield from self.populate_test_engine_data(self.test_environment_info, database_credentials)
+        reuse_databse_setup = not (self.reuse_database_setup and self.test_environment_info.database_info.reused)
+        print("reuse_databse_setup", reuse_databse_setup)
```
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py:70]:
```suggestion
```

File: test/test_run_db_test_docker_db_reuse.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                   "--reuse-test-environment"]
+        command_str = " ".join(command)
+        self.test_environment.run_command(command_str, track_task_dependencies=True)
+        old_test_container_id = self.get_docker_container_id(self._test_container_name)
+        old_db_container_id = self.get_docker_container_id(self._db_container_name)
+        self.test_environment.run_command(command_str, track_task_dependencies=True)
+        new_test_container_id = self.get_docker_container_id(self._test_container_name)
+        new_db_container_id = self.get_docker_container_id(self._db_container_name)
+        self.assertEqual(old_test_container_id, new_test_container_id)
+        self.assertEqual(old_db_container_id, new_db_container_id)
```
Comment (ckunki) [inline test/test_run_db_test_docker_db_reuse.py:46]:
```suggestion
from typing import Dict

    def get_docker_container_ids(self, *names) -> Dict[str, str]:
        result = {}
        with ContextDockerClient() as docker_client:
            for name in names:
                result[name] = docker_client.containers.get(name).id
        return result

    def test_reuse(self):
        def container_ids(command: List[str]) -> Dict[str, str]:
            self.test_environment.run_command(" ".join(command), track_task_dependencies=True)
            return get_docker_container_ids(
                self._test_container_name,
                self._db_container_name,
            )

        command = [f"{self.test_environment.executable}",
                   f"run-db-test",
                   f"{exaslct_utils.get_full_test_container_folder_parameter()}",
                   "--reuse-test-environment"]
        old_ids = container_ids(command)
        new_ids = container_ids(command)
        self.assertEqual(old_ids, new_ids)
```

**Review by ckunki — CHANGES_REQUESTED** [review]

File: exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py [hunk 4]
```diff
@@ -56,43 +56,42 @@ def run_task(self):
...
-        task = self.create_child_task(
-            PopulateTestEngine,
-            test_environment_info=test_environment_info,
-            environment_name=self.test_environment_info.name,
-            db_user=database_credentials.db_user,
-            db_password=database_credentials.db_password,
-            bucketfs_write_password=database_credentials.bucketfs_write_password
-        )
-        yield from self.run_dependencies(task)
+        reuse = not (self.reuse_database_setup and self.test_environment_info.database_info.reused)
```
Comment (ckunki) [inline exasol_script_languages_container_tool/lib/tasks/test/test_runner_db_test_task.py:84]: I still have the feeling that the name `reuse` is wrong as the Assigned boolean condition is wrapped in a `not()`.
Should the task `PopulateTestEngine` be created only if we can **not** reuse the database?
So the logic of the implementation is correct, but the naming is negated and hence contradicting the implementation.

Proposal: 
```python
reuse = self.reuse_database_setup and self.test_environment_info.database_info.reused
if not reuse:
```

File: test/test_run_db_test_docker_db_reuse.py [hunk 5]
```diff
@@ -7,6 +7,16 @@
 
 import utils as exaslct_utils
 
+from typing import Dict, List
+
+
+def get_docker_container_ids(self, *names) -> Dict[str, str]:
```
Comment (ckunki) [inline test/test_run_db_test_docker_db_reuse.py:13]:
```suggestion
def get_docker_container_ids(*names) -> Dict[str, str]:
```

**Review by ckunki — CHANGES_REQUESTED** [review]

File: test/test_run_db_test_docker_db_reuse.py (hunk 5 above)
Comment (tkilias) [reply test/test_run_db_test_docker_db_reuse.py:13]: good catch

**Review by ckunki — APPROVED** [review]

**Pull request merged**